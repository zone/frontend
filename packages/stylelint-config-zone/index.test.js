const { exec } = require('child_process');
const { promisify } = require('util');
const { name: NPM_PACKAGE } = require('./package.json');

const execAsync = promisify(exec);

describe(NPM_PACKAGE, () => {
  it('config does not error', async () => {
    const { stdout } = await execAsync(
      `./node_modules/.bin/stylelint index.scss --printConfig --config=${require.resolve(
        './index.js'
      )}`
    );

    expect(JSON.parse(stdout)).toMatchSnapshot();
  });
});
