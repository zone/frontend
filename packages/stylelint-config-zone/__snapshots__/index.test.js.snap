// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@zonedigital/stylelint-config-zone config does not error 1`] = `
Object {
  "pluginFunctions": Object {},
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/stylelint-prettier/stylelint-prettier.js",
  ],
  "rules": Object {
    "at-rule-empty-line-before": Array [
      "always",
      Object {
        "except": Array [
          "blockless-after-same-name-blockless",
          "first-nested",
        ],
        "ignore": Array [
          "after-comment",
        ],
        "ignoreAtRules": Array [
          "if",
          "else",
        ],
      },
    ],
    "at-rule-name-case": null,
    "at-rule-name-newline-after": null,
    "at-rule-name-space-after": Array [
      "always",
    ],
    "at-rule-no-unknown": null,
    "at-rule-semicolon-newline-after": null,
    "at-rule-semicolon-space-before": Array [
      "never",
    ],
    "block-closing-brace-empty-line-before": null,
    "block-closing-brace-newline-after": Array [
      "always",
      Object {
        "ignoreAtRules": Array [
          "if",
          "else",
        ],
      },
    ],
    "block-closing-brace-newline-before": null,
    "block-closing-brace-space-after": null,
    "block-closing-brace-space-before": null,
    "block-no-empty": Array [
      true,
    ],
    "block-opening-brace-newline-after": null,
    "block-opening-brace-newline-before": null,
    "block-opening-brace-space-after": null,
    "block-opening-brace-space-before": null,
    "color-hex-case": null,
    "color-hex-length": Array [
      "short",
    ],
    "color-no-invalid-hex": Array [
      true,
    ],
    "comment-empty-line-before": Array [
      "always",
      Object {
        "except": Array [
          "first-nested",
        ],
        "ignore": Array [
          "stylelint-commands",
        ],
      },
    ],
    "comment-no-empty": Array [
      true,
    ],
    "comment-whitespace-inside": Array [
      "always",
    ],
    "custom-property-empty-line-before": Array [
      "always",
      Object {
        "except": Array [
          "after-custom-property",
          "first-nested",
        ],
        "ignore": Array [
          "after-comment",
          "inside-single-line-block",
        ],
      },
    ],
    "declaration-bang-space-after": Array [
      "never",
    ],
    "declaration-bang-space-before": Array [
      "always",
    ],
    "declaration-block-no-duplicate-properties": Array [
      true,
      Object {
        "ignore": Array [
          "consecutive-duplicates-with-different-values",
        ],
      },
    ],
    "declaration-block-no-shorthand-property-overrides": Array [
      true,
    ],
    "declaration-block-semicolon-newline-after": null,
    "declaration-block-semicolon-newline-before": null,
    "declaration-block-semicolon-space-after": null,
    "declaration-block-semicolon-space-before": null,
    "declaration-block-single-line-max-declarations": Array [
      1,
    ],
    "declaration-block-trailing-semicolon": null,
    "declaration-colon-newline-after": null,
    "declaration-colon-space-after": null,
    "declaration-colon-space-before": null,
    "declaration-empty-line-before": Array [
      "always",
      Object {
        "except": Array [
          "after-declaration",
          "first-nested",
        ],
        "ignore": Array [
          "after-comment",
          "inside-single-line-block",
        ],
      },
    ],
    "font-family-no-duplicate-names": Array [
      true,
    ],
    "font-family-no-missing-generic-family-keyword": Array [
      true,
    ],
    "function-calc-no-invalid": Array [
      true,
    ],
    "function-calc-no-unspaced-operator": Array [
      true,
    ],
    "function-comma-newline-after": null,
    "function-comma-newline-before": null,
    "function-comma-space-after": null,
    "function-comma-space-before": null,
    "function-linear-gradient-no-nonstandard-direction": Array [
      true,
    ],
    "function-max-empty-lines": null,
    "function-name-case": Array [
      "lower",
    ],
    "function-parentheses-newline-inside": null,
    "function-parentheses-space-inside": null,
    "function-whitespace-after": Array [
      "always",
    ],
    "indentation": null,
    "keyframe-declaration-no-important": Array [
      true,
    ],
    "length-zero-no-unit": Array [
      true,
    ],
    "linebreaks": null,
    "max-empty-lines": null,
    "max-line-length": null,
    "media-feature-colon-space-after": Array [
      "always",
    ],
    "media-feature-colon-space-before": Array [
      "never",
    ],
    "media-feature-name-case": null,
    "media-feature-name-no-unknown": Array [
      true,
    ],
    "media-feature-parentheses-space-inside": Array [
      "never",
    ],
    "media-feature-range-operator-space-after": Array [
      "always",
    ],
    "media-feature-range-operator-space-before": Array [
      "always",
    ],
    "media-query-list-comma-newline-after": null,
    "media-query-list-comma-newline-before": null,
    "media-query-list-comma-space-after": null,
    "media-query-list-comma-space-before": null,
    "no-descending-specificity": Array [
      true,
    ],
    "no-duplicate-at-import-rules": Array [
      true,
    ],
    "no-duplicate-selectors": Array [
      true,
    ],
    "no-empty-first-line": null,
    "no-empty-source": Array [
      true,
    ],
    "no-eol-whitespace": null,
    "no-extra-semicolons": null,
    "no-invalid-double-slash-comments": Array [
      true,
    ],
    "no-missing-end-of-source-newline": null,
    "number-leading-zero": null,
    "number-no-trailing-zeros": null,
    "prettier/prettier": Array [
      true,
    ],
    "property-case": null,
    "property-no-unknown": Array [
      true,
    ],
    "rule-empty-line-before": null,
    "selector-attribute-brackets-space-inside": Array [
      "never",
    ],
    "selector-attribute-operator-space-after": Array [
      "never",
    ],
    "selector-attribute-operator-space-before": Array [
      "never",
    ],
    "selector-attribute-quotes": Array [
      "always",
    ],
    "selector-combinator-space-after": null,
    "selector-combinator-space-before": null,
    "selector-descendant-combinator-no-non-space": null,
    "selector-list-comma-newline-after": null,
    "selector-list-comma-newline-before": null,
    "selector-list-comma-space-after": null,
    "selector-list-comma-space-before": null,
    "selector-max-empty-lines": null,
    "selector-pseudo-class-case": null,
    "selector-pseudo-class-no-unknown": Array [
      true,
    ],
    "selector-pseudo-class-parentheses-space-inside": Array [
      "never",
    ],
    "selector-pseudo-element-case": null,
    "selector-pseudo-element-colon-notation": Array [
      "double",
    ],
    "selector-pseudo-element-no-unknown": Array [
      true,
    ],
    "selector-type-case": Array [
      "lower",
    ],
    "selector-type-no-unknown": Array [
      true,
    ],
    "string-no-newline": Array [
      true,
    ],
    "string-quotes": Array [
      "single",
    ],
    "unicode-bom": null,
    "unit-case": null,
    "unit-no-unknown": Array [
      true,
    ],
    "value-keyword-case": Array [
      "lower",
    ],
    "value-list-comma-newline-after": null,
    "value-list-comma-newline-before": null,
    "value-list-comma-space-after": null,
    "value-list-comma-space-before": null,
    "value-list-max-empty-lines": null,
  },
}
`;
